src/perl/vdotest/VDOTest/ThinVDO.pm

```diff
@@ -0,0 +1,168 @@
+##
+# Basic VDO test using block read/write
+#
+# $Id$
+##
+package VDOTest::ThinVDO;
+
+use strict;
+use warnings FATAL => qw(all);
+use English qw(-no_match_vars);
+use Log::Log4perl;
+use Permabit::Assertions qw(assertNumArgs);
+use Permabit::Constants;
+
+use base qw(VDOTest);
+
+my $log = Log::Log4perl->get_logger(__PACKAGE__);
+
+########################################################################
+# @paramList{getProperties}
+our %PROPERTIES
+  = (
+     # @ple Number of blocks to write
+     blockCount => 5000,
+     # @ple Use a VDO device
+     deviceType => "linear",
+     # @ple VDO slab bit count
+     slabBits   => $SLAB_BITS_SMALL,
+    );
+##
+
+########################################################################
+# Check LVM version.
+##
+sub doesLVMSupportThinVDO {
+  my ($self) = assertNumArgs(1, @_);
+  my $device = $self->getDevice();
+  my $machine = $device->getMachine();
+
+  # Check lvm version. line we want is: 'LVM version:     2.03.26(2) (2024-08-23)'
+  $machine->runSystemCmd("sudo lvm version | grep 'LVM version:' | awk '{print \$3}'");
+  my $stdout = $machine->getStdout();
+  my @version = split(/\(/, $stdout);
+  my ($major, $minor, $patch) = split(/\./, $version[0]);
+  $log->info("LVM version is " . join(".", ($major, $minor, $patch)));
+  if (($major > 2) || ($major == 2 && $minor > 3)
+      || ($major == 2 && $minor == 3 && $patch >= 26)) {
+    return 1;
+  }
+  return 0;
+}
+
+########################################################################
+# Basic VDO testing.
+##
+sub testBasic {
+  my ($self) = assertNumArgs(1, @_);
+
+  if (!$self->doesLVMSupportThinVDO()) {
+    $log->info("LVM version doesn't support thinpool using vdo");
+    return;
+  };
+
+  $self->createTestDevice("thinpoolvdo");
+  $self->createTestDevice("thin");
+
+  my $device = $self->getDevice();
+
+  my $expectedStats = {
+    "data blocks used"       => 0,
+    "dedupe advice valid"    => 0,
+    "dedupe advice stale"    => 0,
+    "dedupe advice timeouts" => 0,
+    "entries indexed"        => 0,
+  };
+  $self->assertVDOStats($expectedStats);
+
+  # Write some blocks, read them back and verify the data have not changed.
+  my $slice1 = $self->createSlice(blockCount => $self->{blockCount});
+  $slice1->write(tag => "Direct1", direct => 1,);
```

Comment:
Trailing comma on this line feel awkward to me. (I know it doesn't do anything functionally.)
Similar comment for lines 88 and 89.
Also at lines 145 and 154.